_posts/2024-01-03-opensearch-performance-improvements.md

```diff
@@ -0,0 +1,1141 @@
+
+## Measured performance improvements
+
+To compare the performance between Elasticsearch 7.10.2 and OpenSearch 2.11, we ran query operations for widely used scenarios in log analytics, time series, and search. We ran the queries across clusters running each version and documented the resulting performance. The following are the key findings from this exercise.
+
+### Log analytics
+
+For log analytics use cases, we used the `http_logs` workload from the [OpenSearch Benchmark](https://opensearch.org/docs/latest/benchmark/index/)---a macro-benchmark utility within the OpenSearch Project---to replicate some of the common query operations. Here are the key highlights:
+
+* `match_all` queries with sorting showed a more than 20x performance boost across the board because of multiple improvements made in the area (see [#6321](https://github.com/opensearch-project/OpenSearch/pull/6321) and [#7244](https://github.com/opensearch-project/OpenSearch/pull/7244)) and other Lucene enhancements.
+
+* Queries for ascending and descending sort-after-timestamp saw a significant performance improvement of up to 70x overall. The optimizations introduced (such as [#6424](https://github.com/opensearch-project/OpenSearch/pull/6424) and [#8167](https://github.com/opensearch-project/OpenSearch/issues/8167)) extend across various numeric types, including but not limited to `int`, `short`, `float`, `double`, `date`, and others.
+
+* Other popular queries such as `search_after` saw an about 60x reduction in latency, attributed to the improvements made in the area involving optimally skipping segments during search (see [#7453](https://github.com/opensearch-project/OpenSearch/pull/7453)). The `search_after` queries can be used as the recommended alternative to scroll queries for a better search experience.
+
+
+### Time series
+
+In the context of aggregations over range and time-series data, we used the `nyc_taxis` and `http_logs` workloads from OpenSearch Benchmark to benchmark various popular use cases. Here are the key highlights:
+
+* Range queries, popular for aggregation use cases, exhibited about 50%--75% improvement, attributed to system upgrades such as Lucene (from v8.8 in Elasticsearch 7.10.2 to v9.7 in OpenSearch 2.11) and JDK (from JDK15 in Elasticsearch 7.10.2 to JDK17 in OpenSearch 2.11).
+
+* Hourly aggregations and multi-term aggregations also demonstrated improvement, varying from 5% to 35%, attributed to similar time-series improvements discussed previously.
+
+* `date_histograms` and `date_histogram_agg` queries exhibited either comparable or slightly decreased performance, ranging from 5% to around 20% in multi-node environments. These issues are actively being addressed as part of the ongoing project efforts (see [#11083](https://github.com/opensearch-project/OpenSearch/pull/11083)).
```

Comment:
Also a line item under Time Series:

* For the date histogram aggregations, there are upcoming changes aiming to improve the performance by rounding-down dates to the nearest interval (such as year, quarter, month, week, day) using SIMD ([#11194](https://github.com/opensearch-project/OpenSearch/pull/11194)).

Reply:
Done